src/rgw/motr/gc/gc_sync.cc

```diff
@@ -0,0 +1,115 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=2 sw=2 expandtab ft=cpp
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * SAL implementation for the CORTX Motr backend
+ *
+ * Copyright (C) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#include "motr/gc/gc_sync.h"
+
+int MotrGCThreadSync::lock(const std::string &lock_name, MotrLockType lock_type, utime_t lock_duration, const std::string locker_id) {
+  assert(motr_gclock);
+  int rc = 0;
+  motr_lock_info_t lock_obj;
+  rc = motr_gclock->read_lock(lock_name, &lock_obj);
+  if (rc == -ENOENT) {
+    // lock is available/free
+  } else {
+    // lock entry is present. Check if this is a stale/expired lock
+    utime_t now = ceph_clock_now();
+    auto iter = lock_obj.lockers.begin();
+    while (iter != lock_obj.lockers.end()) {
+      struct motr_locker_info_t &info = iter->second;
+      if (!info.expiration.is_zero() && info.expiration < now) {
+        // locker has expired; delete it
+        iter = lock_obj.lockers.erase(iter);
+      } else {
+        ++iter;
+      }
+    }
+    if (!lock_obj.lockers.empty()) {
+      // lock is not available
+      return -EBUSY;
+    } else {
+      struct motr_locker_info_t locker;
+      locker.coockie = locker_id;
+      locker.expiration = ceph_clock_now() + lock_duration;
+      locker.description = "";
+      // Update lock entry with current locker and lock the resource
+      lock_obj.lockers.insert(
+          std::pair<std::string, struct motr_locker_info_t>(locker_id,locker));
+      rc = motr_gclock->write_lock(lock_name, &lock_obj, true);
+      if (rc < 0) {
+        // Failed to acquire lock; possibly, already acquired
+      }
```

Comment:
This will be checked by the caller, so we don't need it here or we can log a message.

Reply:
will be used for logging.

Reply:
Add the log statement.